### ISSUE TITLE
Change defValue type to Object

### ISSUE THREAD

**Issue by cr3ativ3** [DroidsOnRoids/jspoon#28]
I'd like to specify default values for other type of fields as well, not just String. Even though there are limitations with what you can set inside an annotation, but being able to set BigDecimal.ZERO or a default value for some Enum field would be helpful.

Ofcourse as long s the defValue type matches the field's type. This could be validated in the HtmlAdapter constructor:
```
if (selector != null) {
     if (!Selector.NO_VALUE.equals(selector.defValue()) && !fieldClass.isAssignableFrom(selector.defValue().getClass()) {
           throw exeption that defValue type does not match field's type
     }
     addCachedHtmlField(field, selector, fieldClass);
}
```

**Comment by devcsrj** [comment]
I encountered this use case too. What I'm hoping for however, is to recognize the _actual_ assigned value in the field. For example:

```
class User {
  @Selector(value = "#username") String name = "anonymous";
}
```

If jspoon can't resolve `#username`, it should instead keep the already assigned value `"anonymous"` instead of `"NO_VALUE"`. The same goes for all other types.

Thoughts?

**Comment by devcsrj** [comment]
Ah, it turns out `defValue` type can't be changed to an `Object` after all, because the [JLS prohibits it](https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.6.10):

> The return type of a method declared in an annotation type must be one of the following, or a compile-time error occurs:
> - A primitive type
> - String
> - Class or an invocation of Class (§4.5)
> - An enum type
> - An annotation type
> - An array type whose component type is one of the preceding types (§10.1).

**Comment by cr3ativ3** [comment]
Looking at the defValue as not a value for the field but rather what is returned from parsing the node, maybe it makes sense its a String.
For the defaulting field's value certainly whatever is instantiated could be used. We should also check if the field is final before any parsing for performance reasons.

**Comment by burnoo** [comment]
Would be released as 1.2.1 soon
